payments/api/views.py

```diff
@@ -120,6 +121,42 @@ def post(self, request, *args, **kwargs):
             return Response(error_data, status=status.HTTP_400_BAD_REQUEST)
 
 
+class ChangeCardTokenView(StripeView):
+    """
+    Add or update customer card token
+
+    This is useful if you are planing to use strip.js to
+    retrieve the card token. This isolates the full credit
+    card number from your server.
+    """
+    serializer_class = CardTokenSerializer
+
+    def post(self, request, *args, **kwargs):
+        try:
+            serializer = self.serializer_class(data=request.data)
+
+            if serializer.is_valid():
+                validated_data = serializer.validated_data
+
+                customer = self.get_customer()
+
+                token = validated_data['token']
+                customer.update_card(token)
```

Comment:
I think we should validate the token first as the underlying `customer.update_card` is not handling incoming error. Previously our own code was generating this token, so there was no need. But now, the token is coming from api. See here https://stripe.com/docs/api#retrieve_token

The below `try.. except stripe.CardError` also won't handle this. The error would be something like:

```python
InvalidRequestError: Request req_8tiPGLdRt: No such token: tok_invalid_test_token
```